crates/ruff_linter/src/rules/pycodestyle/overlong.rs

```diff
@@ -0,0 +1,142 @@
+use std::borrow::Cow;
+
+use unicode_width::UnicodeWidthStr;
+
+use ruff_python_index::Indexer;
+use ruff_python_trivia::is_pragma_comment;
+use ruff_source_file::Line;
+use ruff_text_size::{TextLen, TextRange};
+
+use crate::line_width::{LineLength, LineWidthBuilder, TabSize};
+
+#[derive(Debug)]
+pub(super) struct Overlong {
+    range: TextRange,
+    width: usize,
+}
+
+impl Overlong {
+    /// Returns an [`Overlong`] if the measured line exceeds the configured line length, or `None`
+    /// otherwise.
+    pub(super) fn try_from_line(
+        line: &Line,
+        indexer: &Indexer,
+        limit: LineLength,
+        task_tags: &[String],
+        tab_size: TabSize,
+    ) -> Option<Self> {
+        // The maximum width of the line is the number of bytes multiplied by the tab size (the
+        // worst-case scenario is that the line is all tabs). If the maximum width is less than the
+        // limit, then the line is not overlong.
+        let max_width = line.len() * tab_size.as_usize();
+        if max_width < limit.value() as usize {
+            return None;
+        }
+
+        // Measure the line. If it's already below the limit, exit early.
+        let width = Overlong::measure(line.as_str(), tab_size);
+        if width <= limit {
+            return None;
+        }
+
+        // Strip trailing pragma comments and, if necessary, re-measure the line.
+        let (line, width) = if let Some(line) = Overlong::strip_trailing(line, indexer, task_tags) {
+            // Re-measure the line.
+            let width = Overlong::measure(line.as_str(), tab_size);
+            if width <= limit {
+                return None;
+            }
+
+            (Cow::Owned(line), width)
+        } else {
+            (Cow::Borrowed(line), width)
+        };
```

Comment:
Would it make sense for `strip_trailing` to return the pragma comment instead, considering that the pragma comment tends to be shorter (no indent) than the code? Or is the intention to only highlight the non-pragma part in the violation? (In which case we could return `Option<(&str, &str)>`

Reply:
The intention is to only highlight the non-pragma part of the violation.